src/daemon.rs

```diff
@@ -456,7 +417,7 @@ fn format_message(
     )
 }
 
-pub fn daemon_init(settings: Settings, config: Config) -> Result<Daemon, Error> {
+pub fn daemon_init(settings: Settings, config: Config) -> Result<Arc<Mutex<Daemon>>, Error> {
```

Comment:
just a little cursed lol

Reply:
I honestly don't see a better way to do it though! Do you have any suggestions?

Reply:
Yea, that is fair, it probably is the best solution

Reply:
Here are some good resources I have yet to read, but will soon

- https://www.rustnote.com/concurrency/mutex.html
- https://itsallaboutthebit.com/arc-mutex/

I'm not that great at concurrency in rust yet but I'll do some reading, this seems way less cursed than much of how concurrency happens in languages other than rust, so that's good